apps/devhub/tasks.py

```diff
@@ -32,65 +32,116 @@
 log = logging.getLogger('z.devhub.task')
 
 
+# Simple wrapper for the sake of tests.
+def validate(file_, listed=None):
+    """Run the validator on the given File or FileUpload object, and annotate
+    the results using the ValidationAnnotator."""
+
+    # Import loop.
+    from .utils import ValidationAnnotator
+
+    return ValidationAnnotator(file_, listed=listed).task.delay()
+
+
 @task(soft_time_limit=settings.VALIDATOR_TIMEOUT)
 @write
-def validator(upload_id, listed=True, **kw):
-    if not settings.VALIDATE_ADDONS:
-        return None
+def validate_upload(upload_id, listed, **kw):
+    """Run the validator against a file at the given path, and return the
+    results."""
+    if settings.VALIDATE_ADDONS:
+        upload = FileUpload.objects.get(pk=upload_id)
+        try:
+            result = run_validator(upload.path, listed=listed)
+        except Exception:
+            upload.update(task_error=traceback.format_exc())
+            raise
 
-    log.info('VALIDATING: %s' % upload_id)
-    upload = FileUpload.objects.using('default').get(pk=upload_id)
-    try:
-        result = run_validator(upload.path, listed=listed)
-    except Exception:  # Avoid bare except, which will catch system exceptions.
-        # Store the error with the FileUpload job, then raise
-        # it for normal logging.
-        tb = traceback.format_exception(*sys.exc_info())
-        upload.update(task_error=''.join(tb))
-        raise
+        # FIXME: Have validator return native Python datastructure.
+        return json.loads(result)
```

Comment:
In the previous version, this was surrounded by a `try/except`, don't we need that?